.github/workflows/pr-test.yml

```diff
@@ -0,0 +1,60 @@
+name: test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0
+      
+      - name: create checksum file
+        uses: hypertrace/github-actions/checksum@main
+
+      - name: Cache packages
+        id: cache-packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle
+          key: gradle-packages-${{ runner.os }}-${{ github.job }}-${{ hashFiles('**/checksum.txt') }}
+          restore-keys: |
+            gradle-packages-${{ runner.os }}-${{ github.job }}
+            gradle-packages-${{ runner.os }}
+
+      - name: Unit test
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: jacocoTestReport
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          name: unit test reports
+          fail_ci_if_error: true
+          flags: unit
+  
+      - name: copy test reports
+        uses: hypertrace/github-actions/gradle@main
+        with: 
+          args: copyAllReports --output-dir=/tmp/test-reports
+            
+      - name: Archive test reports
```

Comment:
No real need to archive, the publish step below takes the place of that IMO (since it was just a way to review results)